test/functional/Java8andUp/src/org/openj9/test/string/StringIndexOfString.java

```diff
@@ -0,0 +1,236 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package org.openj9.test.string;
+
+import org.testng.annotations.Test;
+import org.testng.log4testng.Logger;
+import org.testng.Assert;
+import org.testng.AssertJUnit;
+import java.lang.reflect.Field;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.List;
+
+/**
+ * Designed to test SIMD accelerated String.indexOf(String) on platforms such as
+ * the z/Architecture. The vector register size is assumed to be 16 bytes.
+ */
+@Test(groups = { "level.sanity" })
+public class StringIndexOfString {
+
+	public static final Logger logger = Logger.getLogger(StringIndexOfString.class);
+
+	private static final String shortString200Char = "Canada ic *a zzthe xxwwthi xx country located in ...zzthe xxwwthi xx northern part of North Americann. Its ten provinces and three territories extend from the Atlantic to the e xxwwthi xx northern part of North America. Its te Pacific and northward into the Arctic Ocean, covering 9.98 million square kilometres (3.85 million square miles), making it the world's second-largest country by total area. \";\r\n";
```

Comment:
I would make this public and avoid using `final` since the JIT can perform static final field folding and fold away all operations on this string.

Reply:
I changed this string to ``public static`. But static folding isn't the focus of this test nor does it affect the outcome, right?

Reply:
Like all other things in the JIT it will not affect semantics of the user program, however static final field folding may affect what you want to test here, which is a codegen path that the JIT will generate.